### DISCUSSION TITLE
Need Help Gantt - Grafana - InfluxDB

### DISCUSSION THREAD

**Discussion by BigGilo** [marcusolsson/grafana-gantt-panel#18] [Q&A]
Hello,

For starters, sorry for my English (thanks GoogleTranslate) ...

I would need some help, as I cannot get the plugin to work.
I'm trying to create a Gantt with job durations.

Example:
Job name: AccessCritiques
StartTime: 1613462401
EndTime: 1613462551

I am using a powershell script to transfer the data to InfluxDB.
I have to crash in the query because I manage to get the "Task start time" in the "Configure your query".

The query in InfluxDB returns me these values:
> select "StartTime","EndTime" from "AccessCritiques"
name: AccessCritiques
time                StartTime  EndTime
----                ---------  -------
1613558828000000000 1613462401
1613558835000000000            1613462551

I hope I was clear enough and thanks in advance for your help
Gilo

**Comment by marcusolsson** [comment]
> I have to crash in the query because I manage to get the "Task start time" in the "Configure your query".

I have some difficult understanding what you mean here. Are you getting an error, or does the wizard tell you you don't yet have a correct query?

Can you select the Start time field under Dimensions in the panel options? 

If you use Unix epochs as time, they need to be in milliseconds. It looks like yours are in seconds.

**Reply by BigGilo** [reply]
Hi !

When I do my request I only have "StartTime" working
![Gantt](https://user-images.githubusercontent.com/79198976/108229823-703fb100-7140-11eb-90b4-56b73a1a9165.jpg)
![Gantt2](https://user-images.githubusercontent.com/79198976/108231297-e7297980-7141-11eb-9b65-a069de190f2e.jpg)

**Comment by marcusolsson** [comment]
Could you send me the DataFrame structure returned by the data source? You can find it under the panel inspector:

![CleanShot 2021-02-17 at 19 34 46@2x](https://user-images.githubusercontent.com/8396880/108251155-61182d80-7157-11eb-8eb6-9d377a96914b.png)

![CleanShot 2021-02-17 at 19 35 13@2x](https://user-images.githubusercontent.com/8396880/108251134-59f11f80-7157-11eb-9670-509a414e8e19.png)

**Reply by BigGilo** [reply]
Hi !

![Gantt3](https://user-images.githubusercontent.com/79198976/108310753-33150680-71b4-11eb-9938-ff83108c95e2.jpg)

Thx

**Reply by marcusolsson** [reply]
Like the wizard says, you need a string field along with the two time fields, which is most likely why it's not working for you.

StartTime and EndTime are number fields, so they won't be defaulted to, but you can set them explicitly to convert them (as you have). I'm not sure why the second time field isn't showing green though. I'll look into that.

**Reply by marcusolsson** [reply]
Like I thought, if you're using a number or string field as time the wizard incorrectly shows it as missing. I've submitted a fix for this here: https://github.com/marcusolsson/grafana-gantt-panel/commit/d053c7e5841d12a5a3cc1632d279cafcdd3ec0ff

In your case though, the problem is that you haven't specified a string field to use for the labels of each task.

**Reply by BigGilo** [reply]
I think I am not inserting the values ​​correctly in InfluxDB.
I didn't understand which values ​​correspond to which fields.

Are any of these requests correct?


Invoke-RestMethod -Method POST -Uri $InfluxUrl -Body "APXGantt,Job='AccessCritiques',JobName='AccessCritiquesStart' StartTime=1613462401"
Invoke-RestMethod -Method POST -Uri $InfluxUrl -Body "APXGantt,Job='AccessCritiques',JobName='AccessCritiquesEnd' EndTime=1613462551"

Or ...

$JobStartEnd = "1613462401 1613462551"
Invoke-RestMethod -Method POST -Uri $InfluxUrl -Body "APXGantt,Job='AccessCritiques',JobName='AccessCritiquesStartEnd' Time=$JobStartEnd"

Sorry for my lack of knowledge ...

**Reply by marcusolsson** [reply]
No need to apologize for not knowing enough. These things can unfortunately be tricky to get right. If you're struggling, others will too. 

Unfortunately, I don't know enough InfluxDB to tell you whether your queries are correct. Before you start using the Gantt visualization, could you try to configure a query that displays something like this in the Table visualization?

![CleanShot 2021-02-18 at 11 28 10@2x](https://user-images.githubusercontent.com/8396880/108343672-813e0000-71dc-11eb-9f9f-1ccb8d2b7c98.png)

**Reply by BigGilo** [reply]
What type of database do you have there?
Because I don't have the same types of field names

**Reply by marcusolsson** [reply]
I use the [Static](https://grafana.com/plugins/marcusolsson-static-datasource) data source for testing panel plugins. It lets me manually create the data source response in Grafana. It's mostly useful if your testing or prototyping I'd say.
